appscale/tools/appscale_tools.py

```diff
@@ -654,3 +686,145 @@ def upload_app(cls, options):
       shutil.rmtree(file_location)
 
     return (login_host, http_port)
+
+  @classmethod
+  def upgrade(cls, options):
+    """ Upgrades the deployment to the latest AppScale version as well as
+    updates the data.
+    Args:
+      options: A Namespace that has fields for each parameter that can be
+        passed in via the command-line interface.
+    """
+    ips_layout_yaml = yaml.load(options.ips_layout)
+    zk_ips = ips_layout_yaml['zookeeper']
+    db_ips = ips_layout_yaml['database']
+    master_ip = ips_layout_yaml['master']
+    upgrade_version_available = cls.get_upgrade_version_available(master_ip, options.keyname)
+
+    if APPSCALE_VERSION == upgrade_version_available:
+      AppScaleLogger.log("AppScale is already at its latest code version, "
+        "so skipping code pull and build.")
+      AppScaleLogger.log("Running upgrade script to check if any other upgrade is needed.")
+      cls.shut_down_appscale_if_running(options)
```

Comment:
What's the use case for running the upgrade script if the user is already using the latest version?